Sources/Turf/Geometries/Polygon.swift

```diff
@@ -297,4 +297,55 @@ extension Polygon {
             ring[2].longitude == ring[0].longitude
         )
     }
-}
+
+
+    /// Calculates the centroid using the mean of all vertices.
+    /// This lessens the effect of small islands and artifacts when calculating the centroid of a set of polygons.
+    public func centroid() -> LocationCoordinate2D? {
+        let coordinates = outerRing.coordinates.dropLast()
```

Comment:
In the future, once similar calculations are implemented for other geometry types, Geometry itself could have `centroid` and `centerOfMass` properties that pass through to type-specific implementations like this.